### PR TITLE
Add Date A Portlet Rating Was Done

### PR DESCRIPTION
**Pull request by phillips1021**
##### Checklist

-   [ X] the [individual contributor license agreement][] is signed
-   [ X] commit message follows [commit guidelines][]
-   [ X] tests are included

##### Description of change

Add date a portlet rating was completed

https://issues.jasig.org/browse/UP-4621

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: uPortal-web/src/main/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDao.java [hunk 1]
```diff
@@ -302,4 +305,53 @@ public void aggregateMarketplaceRating() {
             }
         }
     }
+    
+    @Override
+    public Set<IMarketplaceRating> getAllRatingsForPortletInLastXDays(int numberOfDaysBack, final String fname) {
+        
+        final Calendar queryDate = Calendar.getInstance();
+        queryDate.add(Calendar.DAY_OF_YEAR, 0-numberOfDaysBack);
```
Comment (ChristianMurphy) [inline uPortal-web/src/main/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDao.java]: :question: Could the Java 8 Time processing utils be used to simplify/clarify the code?
http://www.baeldung.com/migrating-to-java-8-date-time-api

File: uPortal-webapp/src/test/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDaoTest.java [hunk 2]
```diff
@@ -279,4 +284,121 @@ public Object call() throws Exception {
...
+                        int starRating = random.nextInt(MarketplaceRatingImpl.MAX_RATING) + 1;
+                        rating.setRating(starRating);
+                        MarketplaceRatingPK ratingPK = new MarketplaceRatingPK();
+                        ratingPK.setPortletDefinition((PortletDefinitionImpl) portlet);
+                        ratingPK.setUserName(person.getName());
+                        rating.setMarketplaceRatingPK(ratingPK);
+                        rating.setRatingDate(today.getTime());
+                        rating.setReview("comment for " + ratingPK.getPortletDefinition().getFName());
+                        marketplaceRatingDao.createOrUpdateRating(rating);
+                        i = i + 20;
```
Comment (ChristianMurphy) [inline uPortal-webapp/src/test/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDaoTest.java]: :question: what is `i + 20` doing? Is it intended?

Comment (phillips1021) [reply uPortal-webapp/src/test/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDaoTest.java]: For the test I want to create portlet ratings with different past dates so i represents how many days in the past (from Today) that the test should use to calculate the rating date.

File: uPortal-web/src/main/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDao.java (hunk 1 above)
Comment (phillips1021) [reply uPortal-web/src/main/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDao.java]: You're correct could use LocalDateTime minusDays method:
https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#minusDays-long- 

not sure what to do to modify this given the submited pull request

File: uPortal-webapp/src/test/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDaoTest.java (hunk 2 above)
Comment (ChristianMurphy) [reply uPortal-webapp/src/test/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDaoTest.java]: :question: Could `i` be named something `numberOfDaysBack` to give some additional clarity on the purpose?
:question:  maybe `20` could become something like `final int TEST_DATE_INCREMENT = 20`?

Comment (phillips1021) [reply uPortal-webapp/src/test/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDaoTest.java]: sure I can make these changes - but not sure where to make the change so it ends up in this pull request

**Comment by phillips1021** [comment]
I'm going to make changes and resubmit with a different branch name

File: uPortal-web/src/main/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDao.java (hunk 1 above)
Comment (ChristianMurphy) [reply uPortal-web/src/main/java/org/apereo/portal/portlet/dao/jpa/JpaMarketplaceRatingDao.java]: > to modify this given the submited pull request

When more changes are pushed to a git branch, those changes are automatically included in a pull request.

**Pull request closed without merging**